tests/integration-tests/tests/test_raid.py

```diff
@@ -0,0 +1,85 @@
+def test_raid_performance_mode(scheduler, pcluster_config_reader, clusters_factory):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    mount_dir = "/raid_dir"
+    _test_raid_correctly_configured(remote_command_executor, raid_type="0", volume_size=75, raid_devices=5)
+    _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size=74)
+    _test_raid_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+
+
+@pytest.mark.regions(["us-east-1", "eu-west-1", "cn-north-1", "us-gov-west-1"])
+@pytest.mark.instances(["c5.xlarge"])
+@pytest.mark.schedulers(["sge", "awsbatch"])
+@pytest.mark.usefixtures("region", "os", "instance")
+def test_raid_fault_tolerance_mode(scheduler, pcluster_config_reader, clusters_factory):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    mount_dir = "/raid_dir"
+    _test_raid_correctly_configured(remote_command_executor, raid_type="1", volume_size=20, raid_devices=2)
+    _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size=20)
+    _test_raid_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+
+
+def _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size):
+    logging.info("Testing raid {0} is correctly mounted".format(mount_dir))
+    result = remote_command_executor.run_remote_command(
+        "df -h -t ext4 | tail -n +2 | awk '{{print $2, $6}}' | grep '{0}'".format(mount_dir)
+    )
+    assert_that(result.stdout).matches(r"{size}G {mount_dir}".format(size=volume_size, mount_dir=mount_dir))
+
+    result = remote_command_executor.run_remote_command("cat /etc/fstab")
+    assert_that(result.stdout).matches(
+        r"/dev/md0 {mount_dir} ext4 defaults,nofail,_netdev 0 2".format(mount_dir=mount_dir)
+    )
+
+
+def _test_raid_correctly_shared(remote_command_executor, mount_dir, scheduler_commands):
+    logging.info("Testing raid correctly mounted on compute nodes")
+    verify_directory_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+
+
+def _test_raid_correctly_configured(remote_command_executor, raid_type, volume_size, raid_devices):
+    result = remote_command_executor.run_remote_command("sudo mdadm --detail /dev/md0")
+    assert_that(result.stdout).contains("Raid Level : raid{0}".format(raid_type))
+    assert_that(result.stdout).matches(r"Array Size : .*{0}\.00 GiB".format(volume_size))
+    assert_that(result.stdout).contains("Raid Devices : {0}".format(raid_devices))
+    assert_that(result.stdout).contains("Raid Devices : {0}".format(raid_devices))
```

Comment:
You have a duplicate row, is this correct?

Reply:
adding some redundancy in test assertions :D

It was supposed to have Active Devices. Fixed, thanks!